packages/core/src/sink/IndexSink.ts

```diff
@@ -0,0 +1,41 @@
+/** @license MIT License (c) copyright 2010-2016 original author or authors */
+/** @author Brian Cavalier */
+/** @author John Hann */
+
+import Pipe from './Pipe'
+import { Time, Sink } from '@most/types'
+
+export interface IndexedValue<A> {
+  readonly index: number
+  readonly value: A
+  readonly active: boolean
+}
+
+export default class IndexSink<A> extends Pipe<A, Readonly<IndexedValue<A | undefined>>> implements Sink<A> {
```

Comment:
What’s the benefit of `Readonly<IndexedValue<A | undefined>>` when `IndexedValue<A>`’s properties already are marked as `readonly` in the interface declaration?

Reply:
fixed